src/Compilers/CSharp/Portable/Binder/Binder_Await.cs

```diff
@@ -115,8 +116,12 @@ private bool CouldBeAwaited(BoundExpression expression)
                 return false;
             }
 
-            var boundAwait = BindAwait(expression, expression.Syntax, BindingDiagnosticBag.Discarded);
-            return !boundAwait.HasAnyErrors;
+            // Could we bind await on this expression (ignoring whether we are in async context)?
+            var syntax = expression.Syntax;
+            var placeholder = new BoundAwaitableValuePlaceholder(syntax, expression.Type);
+            bool hasErrors = ReportBadAwaitContext(syntax, syntax.Location, BindingDiagnosticBag.Discarded);
+            _ = BindAwaitInfo(placeholder, syntax, BindingDiagnosticBag.Discarded, ref hasErrors, expressionOpt: expression);
```

Comment:
> do you think it would be better to call ```BindAwait``` and pass an additional flag to skip the async context check?

I think it is better to have logic that doesn't check content of a DiagnosticBag as long as the logic is sufficiently reliable